src/utils/event-manager.js

```diff
@@ -0,0 +1,226 @@
+import {
+  Manager,
+  Tap,
+  Press,
+  Pinch,
+  Rotate,
+  Pan,
+  Swipe
+} from 'hammerjs';
+import PointerMoveEventInput from './pointer-move-event-input';
+import WheelInput from './wheel-input';
+
+/**
+ * Only one set of basic input events will be fired by Hammer.js:
+ * either pointer, touch, or mouse, depending on system support.
+ * In order to enable an application to be agnostic of system support,
+ * alias basic input events into "classes" of events: down, move, and up.
+ * See `_onBasicInput()` for usage of these aliases.
+ */
+/* eslint-disable */
+const BASIC_EVENT_CLASSES = {
+  down: ['pointerdown', 'touchstart', 'mousedown'],
+  move: ['pointermove', 'touchmove',  'mousemove'],
+  up:   ['pointerup',   'touchend',   'mouseup']
+};
+/* eslint-enable */
+
+const BASIC_EVENT_ALIASES = {
+  pointerdown: BASIC_EVENT_CLASSES.down,
+  pointermove: BASIC_EVENT_CLASSES.move,
+  pointerup: BASIC_EVENT_CLASSES.up,
+  touchstart: BASIC_EVENT_CLASSES.down,
+  touchmove: BASIC_EVENT_CLASSES.move,
+  touchend: BASIC_EVENT_CLASSES.up,
+  mousedown: BASIC_EVENT_CLASSES.down,
+  mousemove: BASIC_EVENT_CLASSES.move,
+  mouseup: BASIC_EVENT_CLASSES.up
+};
+
+/**
+ * "Gestural" events are those that have semantic meaning beyond the basic input event,
+ * e.g. a click or tap is a sequence of `down` and `up` events with no `move` event in between.
+ * Hammer.js handles these with its Recognizer system;
+ * this block maps event names to the Recognizers required to detect the events.
+ */
+const EVENT_RECOGNIZER_MAP = {
+  click: 'tap',
+  tap: 'tap',
+  doubletap: 'tap',
+  press: 'press',
+  pinch: 'pinch',
+  pinchin: 'pinch',
+  pinchout: 'pinch',
+  pinchstart: 'pinch',
+  pinchmove: 'pinch',
+  pinchend: 'pinch',
+  pinchcancel: 'pinch',
+  rotate: 'rotate',
+  rotatestart: 'rotate',
+  rotatemove: 'rotate',
+  rotateend: 'rotate',
+  rotatecancel: 'rotate',
+  pan: 'pan',
+  panstart: 'pan',
+  panmove: 'pan',
+  panup: 'pan',
+  pandown: 'pan',
+  panleft: 'pan',
+  panright: 'pan',
+  panend: 'pan',
+  pancancel: 'pan',
+  swipe: 'swipe',
+  swipeleft: 'swipe',
+  swiperight: 'swipe',
+  swipeup: 'swipe',
+  swipedown: 'swipe'
+};
+
+const RECOGNIZERS = {
+  doubletap: new Tap({
+    event: 'doubletap',
+    taps: 2
+  }),
+  pan: new Pan({
+    threshold: 10
+  }),
+  pinch: new Pinch(),
+  press: new Press(),
+  rotate: new Rotate(),
+  swipe: new Swipe(),
+  tap: new Tap()
+};
+
+/**
+ * Map gestural events typically provided by browsers
+ * that are not reported in 'hammer.input' events
+ * to corresponding Hammer.js gestures.
+ */
+const GESTURE_EVENT_ALIASES = {
+  tap: 'click'
+};
+
+/**
+ * Single API for subscribing to events about both
+ * basic input events (e.g. 'mousemove', 'touchstart', 'wheel')
+ * and gestural input (e.g. 'click', 'tap', 'panstart').
+ * Delegates event registration and handling to Hammer.js.
+ */
+class EventManager {
+  constructor(element, options) {
```

Comment:
This file would be much easier to review if there was a README.md containing an initial version of the documentation for this class, documenting options etc for this class.